### PR TITLE
bug/#157283077/add error message to signup page

### PR DESCRIPTION
**Pull request by Spectrumsun**
#### What does this PR do?
add error handling to user signup page
add image preview to the signup page

#### Description of Task to be completed?
a user should get an error message if the username or email is already used
a preview of the image is display on the page

#### How should this be manually tested?
on the signup page, the user will get an error message if the email or password is already used

#### Any background context you want to provide?
A user doesn't get any notification or error if the username or email is already used. make it a confusing for the user

#### What are the relevant pivotal tracker stories?
#157283077

#### Screenshots (if appropriate)
none

#### Questions:
none

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -225,12 +225,12 @@ exports.checkUsername = (req, res, next) => {
 exports.avatars = function (req, res) {
   // Update the current user's profile to include the avatar choice they've made
   if (req.user && req.user._id && req.body.avatar !== undefined &&
-    /\d/.test(req.body.avatar) && avatars[req.body.avatar]) {
+    /\d/.test(req.body.avatar) && avatars[req.body.avatar]) {  // eslint-disable-line
```
Comment (chuzksy-codeactive) [inline app/controllers/users.js]: Good job on this part. It shows that you are paying attention to details.

File: app/controllers/users.js [hunk 2]
```diff
@@ -225,12 +225,12 @@ exports.checkUsername = (req, res, next) => {
...
     User.findOne({
       _id: req.user._id
     })
       .exec((err, user) => {
-        user.avatar = avatars[req.body.avatar];
+        user.avatar = avatars[req.body.avatar];  // eslint-disable-line
```
Comment (chuzksy-codeactive) [inline app/controllers/users.js]: Same here, you responded in time to fix it. Thumbs up

**Review by chuzksy-codeactive — COMMENTED** [review]
I think we are good here. Nice job

**Review by kenoseni — COMMENTED** [review]
Well done for adding this to the signup page, I can see that there is a merge conflict between this branch and staging. Please respond to this.

**Pull request merged**